app/src/main/java/com/randomappsinc/studentpicker/editing/EditNameListFragment.java

```diff
@@ -195,10 +196,8 @@ public void onDeletionSubmitted(String name, int amountToDelete) {
     }
 
     @Override
-    public void onDuplicationSubmitted(String name, int amountToAdd) {
-        nameListDataManager.addName(getContext(), name, amountToAdd, listName);
-        namesAdapter.addNames(name, amountToAdd);
-        UIUtils.showSnackbar(parent, R.string.clones_added);
+    public void onNameAmountAdjustmentSubmitted(String name, int amountToHave, int currentAmount) {
+       nameListDataManager.setNameAmount(name, amountToHave, currentAmount, getActivity(), listName);
```

Comment:
I think we can just show a Snackbar here instead of invoking the NameListDataManager and then having it communicate back to us. Also, similar to how we exposed a new API in NameListDataManager, we should just expose a new API in namesAdapter as well. We can also clean up the old purely additive APIs along the way as well.